tests/consistency.rs

```diff
@@ -39,6 +39,109 @@ fn compare_birds() {
     compare_audio(&audio);
 }
 
+#[test]
+#[traced_test]
+fn include_unprocessed_bird() {
+    let audio = Path::new("tests/audio/birds.wav");
+    for chunk_ms in [20, 30, 50] {
+        should_include_unprocessed_when_is_speaking(audio, chunk_ms);
+    }
+}
+
+#[test]
+#[traced_test]
+fn include_unprocessed_audio_1() {
+    let audio = Path::new("tests/audio/sample_1.wav");
+    for chunk_ms in [20, 30, 50] {
+        should_include_unprocessed_when_is_speaking(audio, chunk_ms);
+    }
+}
+
+#[test]
+#[traced_test]
+fn include_unprocessed_audio_2() {
+    let audio = Path::new("tests/audio/sample_2.wav");
+    for chunk_ms in [20, 30, 50] {
+        should_include_unprocessed_when_is_speaking(audio, chunk_ms);
+    }
+}
+
+#[test]
+#[traced_test]
+fn include_unprocessed_audio_3() {
+    let audio = Path::new("tests/audio/sample_3.wav");
+    for chunk_ms in [20, 30, 50] {
+        should_include_unprocessed_when_is_speaking(audio, chunk_ms);
+    }
+}
+
+#[test]
+#[traced_test]
+fn include_unprocessed_audio_4() {
+    let audio = Path::new("tests/audio/sample_4.wav");
+    for chunk_ms in [20, 30, 50] {
+        should_include_unprocessed_when_is_speaking(audio, chunk_ms);
+    }
+}
+
+/// When is_speaking is true, get_current_speech should return everything starting from
+/// speech start time until the very end, including the unprocessed samples.
+fn should_include_unprocessed_when_is_speaking(audio: &Path, chunk_ms: usize) {
+    let config = VadConfig::default();
+    let mut vad = VadSession::new(config).unwrap();
+
+    // Read audio samples.
+    let step = if config.sample_rate == 16000 {
+        1
+    } else {
+        2 // Other sample rates are invalid so we'll just work on less data
+    };
+    let samples: Vec<f32> = WavReader::open(audio)
+        .unwrap()
+        .into_samples()
+        .step_by(step)
+        .map(|x| {
+            let modified = x.unwrap_or(0i16) as f32 / (i16::MAX as f32);
+            modified.clamp(-1.0, 1.0)
+        })
+        .collect();
+
+    // Send each chunk to vad for inference.
+    let chunk_size = (config.sample_rate * chunk_ms) / 1000;
+    let num_chunks = samples.len() / chunk_size;
+    let mut last_speech_start_ms = 0;
+    for i in 0..num_chunks {
+        let start = i * chunk_size;
+        let end = if i < num_chunks - 1 {
+            start + chunk_size
+        } else {
+            samples.len()
+        };
+
+        let transitions = vad.process(&samples[start..end]).unwrap();
+        // Update last_speech_start_ms
+        for transition in transitions {
+            if let VadTransition::SpeechStart { timestamp_ms } = transition {
+                last_speech_start_ms = timestamp_ms;
+            }
+        }
+
+        // The actual test logic.
+        if vad.is_speaking() {
+            let current_speech = vad.get_current_speech();
```

Comment:
Do we know that this branch is ever exercised. Because if it isn't we'll be adding a test that does nothing which isn't too useful.

Indeed birds has no parts where it shows as `is_speaking` so that test does nothing :eyes: